Source/Scene/Cesium3DTilesetAsyncTraversal.js

```diff
@@ -0,0 +1,141 @@
+define([
+        '../Core/Intersect',
+        '../Core/ManagedArray',
+        './Cesium3DTileRefine'
+    ], function(
+        Intersect,
+        ManagedArray,
+        Cesium3DTileRefine) {
+    'use strict';
+
+    /**
+     * Traversal that loads all leaves that intersect the camera frustum.
+     * Used to determine ray-tileset intersections during a pickFromRayMostDetailed call.
+     *
+     * @private
+     */
+    function Cesium3DTilesetAsyncTraversal() {
```

Comment:
Does this work with tilesets of tilesets?  Does it need to?

Reply:
Shouldn't this have unit tests?

Reply:
I still need to write tests for 3D Tiles. But this does work with external tilesets. The traversal is not considered ready until all the leaf tiles that are hit contain content.